src/components/TagsManager.vue

```diff
@@ -104,34 +113,46 @@
-                                :class="{'is-invalid': validateDraftTag.valueInvalid}"
+                                :class="{'is-invalid': validateDraftTag.invalid && validateDraftTag.messageKey === 'tagAlreadyOnMonitor'}"
                                 :placeholder="$t('value (optional)')"
                                 data-testid="tag-value-input"
                                 @keydown.enter.prevent="onEnter"
                             />
-                            <div class="invalid-feedback">
-                                {{ $t("Tag with this value already exist.") }}
-                            </div>
                         </div>
-                        <div class="mb-2">
-                            <button
-                                type="button"
-                                class="btn btn-secondary float-end"
-                                :disabled="processing || validateDraftTag.invalid"
-                                data-testid="tag-submit-button"
-                                @click.stop="addDraftTag"
-                            >
-                                {{ $t("Add") }}
-                            </button>
+
+                        <!-- General feedback area for tag input validation messages -->
+                        <div v-if="validateDraftTag.invalid && validateDraftTag.messageKey && (newDraftTag.select != null || canStageMoreNewSystemTags || validateDraftTag.messageKey !== 'tagLimitReached')" class="form-text text-danger mb-2">
+                            {{ $t(validateDraftTag.messageKey, validateDraftTag.messageParams) }}
+                        </div>
+                        <!-- End Validation Feedback -->
+
+                        <!-- Add Another Tag Button -->
+                        <!-- This button will be moved to the footer -->
+
+                        <!-- Action Buttons: Clear current form and Stage current tag -->
+                        <!-- Removed original action buttons here -->
+                        <div v-if="newDraftTag.select == null && !canStageMoreNewSystemTags && validateDraftTag.invalid && validateDraftTag.messageKey === 'tagLimitReached'" class="form-text text-danger text-end mt-1">
+                            {{ $t(validateDraftTag.messageKey, validateDraftTag.messageParams) }}
                         </div>
                     </div>
+                    <!-- Modal Footer: Cancel batch or Confirm and Add all staged tags -->
+                    <div class="modal-footer">
+                        <button type="button" class="btn btn-outline-primary me-2" :disabled="processing || validateDraftTag.invalid" @click.stop="stageCurrentTag">
+                            {{ $t("Add Another Tag") }}
+                        </button>
+                        <button type="button" class="btn btn-secondary" @click.stop="clearStagingAndCloseModal">{{ $t("Cancel") }}</button>
```

Comment:
not sure, but I think if we order it cancel, add another´, add all this seems more "natural" to me.
Not sure why, likely just being used to this.

```suggestion
                        <button type="button" class="btn btn-secondary" @click.stop="clearStagingAndCloseModal">{{ $t("Cancel") }}</button>
                        <button type="button" class="btn btn-outline-primary me-2" :disabled="processing || validateDraftTag.invalid" @click.stop="stageCurrentTag">
                            {{ $t("Add Another Tag") }}
                        </button>
```